magda-web-client/src/helpers/ChartDatasetEncoder.js

```diff
@@ -12,47 +12,145 @@ import takeRight from "lodash/takeRight";
 import sortBy from "lodash/sortBy";
 import * as d3 from "d3-collection";
 import { config } from "../config";
+import chrono from "chrono-node";
 import type { ParsedDistribution } from "../helpers/record";
 
-let Papa = null;
-
-const avlChartTypes = ["bar", "pie", "scatter", "line"];
-const fetchData = function(url) {
-    return new Promise((resolve, reject) => {
-        Papa.parse(config.proxyUrl + "_0d/" + url, {
-            download: true,
-            header: true,
-            skipEmptyLines: true,
-            trimHeader: true,
-            complete: results => {
-                resolve(results);
-            },
-            error: err => {
-                let e;
-                if (!err)
-                    e = new Error("Failed to retrieve or parse the file.");
-                else e = err;
-                reject(e);
-            }
-        });
+const AVAILABLE_CHART_TYPES = ["bar", "pie", "scatter", "line"];
+const STRIP_NUMBER_REGEX = /[^\-\d.+]/g;
+
+/** Columns with these names are more likely to be picked as the default column on the x axis */
+const HIGH_PRIORITY_X_AXES = [
+    "gender",
+    "sex",
+    "occupation",
+    "state",
+    "city",
+    "company",
+    "postcode",
+    "category"
+];
+/** Columns with these names will not be picked as the default column on the x axis unless there's nothing else */
+const LOW_PRIORITY_X_AXES = ["name"];
+/** Columns with these names are more likely to be picked as the default column on the y axis */
+const HIGH_PRIORITY_Y_AXES = [];
+/** Columns with these names will not be picked as the default column on the y axis unless there's nothing else */
+const LOW_PRIORITY_Y_AXES = ["count", "id"];
+
+/** Columns with these terms in them will be overridden as category types */
+const OVERRIDE_TO_CATEGORY_COLUMNS = [
+    "id",
+    "code",
```

Comment:
- Probably we want to keep `age` as category columns? ---- it likely a user wants to see other data according to different ages?
- `lic num` probably is another one we want to keep as category columns? --- otherwise, the result will be license number added up. 
- "longitude", "latitude" probably should be treated as category columns (it won't produce any meaningful result by adding the value up)?  However, move them to xAxis could also likely lead to too many columns (or very busy chart). Probably, another option is to remove those columns from both `xAxis` & `yAxis`. i.e. Exclude them from the chart as they won't be useful for either `xAxis` & `yAxis` columns?

Reply:
- Yeah good point re: age as a category. I've kept age as a numeric for now because given the choice between being able to have age on the Y axis or the X axis, I prefer the Y axis (so you can graph age by category, for instance) but we should make it so you can put numeric columns on either axis at some point.
- Added `lic` to overrides for category columns
- Filtered out `lat` and `long` from all axes.